homeassistant/components/neevo/sensor.py

```diff
@@ -0,0 +1,91 @@
+"""Support for Nee-Vo Tank Monitors."""
+
+from __future__ import annotations
+
+from contextlib import suppress
+import logging
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfPressure
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import NeeVoEntity
+from .const import COORDINATOR, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="level",
+        name="Tank Level",
+        icon="mdi:propane-tank",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="tank_last_pressure",
+        name="Tank Last Pressure",
+        icon="mdi:gauge",
+        native_unit_of_measurement=UnitOfPressure.KPA,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Nee-Vo sensor based on a config entry."""
+
+    instance = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = [
+        NeeVoSensor(instance, tank_id, description)
+        for tank_id, _equip in instance[COORDINATOR].data.items()
+        for description in SENSOR_TYPES
+        if getattr(_equip, description.key, False) is not False
+    ]
+
+    async_add_entities(sensors)
+
+
+class NeeVoSensor(NeeVoEntity, SensorEntity):
```

Comment:
Some points I believe might be missing on this class:

* Annotate with `device_class` for these sensors (unit of measurement is already implemented)
* In case of being a numerical sensor, stating the `StateClass` so statistics can be properly stored for this sensor.
* Adding tests for sensors created by this integration (this should increase the test coverage)

Reply:
Not sure what you mean with annotate with `device_class`
Also not sure what you mean by stating the `StateClass`
Don't actully know how to test sensors that are created by/use an API. All the integrations I know that do a similar thing dont have tests on their sensors so I don't know where to look for examples. Any suggestions.

Reply:
Sorry. Been a while since I've worked on this. State class is set in the sensor descriptions. I'm still not sure about device_class...